src/libs/OptimisticReportNames.ts

```diff
@@ -212,10 +216,10 @@ function computeReportNameIfNeeded(report: Report | undefined, incomingUpdate: O
             // Checking if the formula part is affected in this manner works, but it could certainly be more precise.
             // For example, a policy update only affects the part if the formula in the policy changed, or if the report part references a field on the policy.
             // However, if we run into performance problems, this would be a good place to optimize.
-            return updateType === 'report' || updateType === 'transaction' || updateType === 'policy';
+            return updateType === 'report' || updateType === 'transaction' || updateType === 'policy' || updateType === 'reportNameValuePairs';
```

Comment:
NAB: Let's please add some constants at the top of this file for each update type.